src/components/common/Select.js

```diff
@@ -130,3 +132,65 @@ export const AsyncCreatableSelect = props => {
     ...props
   })
 }
+
+const VirtualizedMenuList = props => {
+  const { children, focusedOption, getStyles, getValue, maxHeight } = props
+
+  const list = useRef()
+
+  // Scroll to the currently selected value (if there is one) when the menu is opened.
+  useOnMount(() => {
+    const [value] = getValue()
+    if (value) {
+      const renderedOptions = Children.map(children, child => child.props.data)
+      const valueIndex = renderedOptions.findIndex(opt => opt.value === value.value)
+      if (valueIndex !== -1) {
+        list.current.scrollToRow(valueIndex)
+      }
+    }
+  })
+
+  // When navigating option with arrow keys, scroll the virtualized list so that the focused option is visible.
+  useEffect(() => {
+    const renderedOptions = Children.map(children, child => child.props.data)
+    const focusedOptionIndex = renderedOptions.indexOf(focusedOption)
+    if (focusedOptionIndex !== -1) {
+      list.current.scrollToRow(focusedOptionIndex)
+    }
+  }, [children, focusedOption])
+
+  const hasRenderedOptions = Array.isArray(children)
+
+  // If no options are rendered, then render the "no options" message.
+  if (!hasRenderedOptions) {
+    return children
+  }
+
+  const rowCount = children.length
+  const rowHeight = 40
```

Comment:
Why do we have to specify the row height vs. it being determined by the rendered list item?

Reply:
react-virtualized requires it: https://github.com/bvaughn/react-virtualized/blob/master/docs/List.md#prop-types. I think so that it can handle scrolling within a virtualized list.